ygot/render.go

```diff
@@ -37,19 +36,86 @@ const (
 // path stores the elements of a path for a particular leaf,
 // such that it can be used as a key for maps.
 type path struct {
-	p []interface{}
+	p *pathWrapper
+}
+
+// GNMIPathFormat specifies the type of paths that are to be
+// used in gNMI.
+type GNMIPathFormat int64
```

Comment:
I made this change, but I'm not 100% happy here since it seems a little fragile to me that `[]string{} != nil` infers quite so much through the code. Isn't an explicit signal here to the code somewhat more robust? It also helps handle the case whereby a prefix that is supplied is explicitly meant to be `nil`.  thoughts?

Reply:
I wouldn't think so, provided that you treat this type as a class rather than simple struct (i.e. use methods rather than direct field access). This should be robust assuming that both path types cannot be set at once (which you now check in isValid). If you need to check if a path is nil, I recommend adding a method called isNil rather than checking directly. Then the caller doesn't even have to worry about what type it is (unless they want to and then they can do p.isStringSlicePath() && p.isNil() for example).